response_operations_ui/views/messages.py

```diff
@@ -48,69 +49,53 @@ def create_message():
                            breadcrumbs=breadcrumbs)
 
 
-def _build_create_message_breadcrumbs():
-    return [
-        {"title": "Messages", "link": "/messages"},
-        {"title": "Create Message"}
-    ]
-
-
-def _get_conversation_breadcrumbs(messages):
-    return [
-        {"title": "Messages", "link": "/messages"},
-        {"title": messages[-1].get('subject', 'No Subject')}
-    ]
-
-
-def _repopulate_form_with_submitted_data(form):
-    form.survey.text = form.hidden_survey.data
-    form.ru_ref.text = form.hidden_ru_ref.data
-    form.business.text = form.hidden_business.data
-    form.to.text = form.hidden_to.data
-    form.body.data = g.form_body_data
-    form.subject.data = g.form_subject_data
-    return form
-
+@messages_bp.route('/threads/<thread_id>', methods=['GET', 'POST'])
+@login_required
+def view_conversation(thread_id):
+    try:
+        thread_conversation = message_controllers.get_conversation(thread_id)['messages']
+        refined_thread = [_refine(message) for message in reversed(thread_conversation)]
+        breadcrumbs = _get_conversation_breadcrumbs(thread_conversation)
+    except KeyError as e:
+        logger.exception("A key error occurred")
+        raise ApiError(e)
+    except IndexError:
+        breadcrumbs = [
+            {"title": "Messages", "link": "/messages"},
+            {"title": "Unavailable"}
+        ]
 
-def _get_message_json(form):
-    return json.dumps({
-        'msg_from': current_user.id,
-        'msg_to': [form.hidden_to_uuid.data],
-        'subject': form.subject.data,
-        'body': form.body.data,
-        'thread_id': "",
-        'collection_case': "",
-        # TODO Make this UUID for v2 api
-        'survey': form.hidden_survey_id.data,
-        'ru_id': form.hidden_to_ru_id.data})
+    if refined_thread[-1]['unread']:
```

Comment:
Should there be some error handling here?  It'll throw a keyError if unread isn't defined.  Or can we basically guarantee that we'll have what we want here?

Reply:
From what I remember, the `_refine` function catches any errors?

Reply:
just took a looksee, it does indeed.